packages/engineer/test/gui.unit.ts

```diff
@@ -0,0 +1,96 @@
+import { createDisposables, RuntimeFeature, BaseHost } from '@wixc3/engine-core/src';
+import { createBrowserProvider } from '@wixc3/engine-test-kit/src';
+import { runNodeEnvironment, loadFeaturesFromPackages, resolvePackages } from '@wixc3/engine-scripts/src';
+import fs from '@file-services/node';
+import devServerFeature, { devServerEnv } from '../feature/dev-server.feature';
+import guiFeature from '../feature/gui.feature';
+import { waitFor } from 'promise-assist';
+import { expect } from 'chai';
+import type { Page } from 'puppeteer';
+
+function getBodyContent(page: Page) {
+    return page.evaluate(() => document.body.textContent!.trim());
+}
+
+describe('engineer:gui', function () {
+    this.timeout(15_000);
+    const disposables = createDisposables();
+    const browserProvider = createBrowserProvider();
+
+    const setup = async ({ featureName, basePath }: { featureName?: string; basePath: string }) => {
+        const features = loadFeaturesFromPackages(resolvePackages(__dirname + '../'), fs).features;
+        const { dispose, engine } = await runNodeEnvironment({
+            featureName: 'engineer/gui',
+            features: [...features],
+            name: devServerEnv.env,
+            type: 'node',
+            host: new BaseHost(),
+            config: [
+                devServerFeature.use({
+                    devServerConfig: {
+                        basePath,
+                    },
+                }),
+                guiFeature.use({
+                    engineerConfig: {
+                        features,
+                    },
+                }),
+            ],
+        });
+
+        const runtimeFeature = engine.features.get(guiFeature) as RuntimeFeature;
+        const devServerRuntime = engine.features.get(devServerFeature) as RuntimeFeature;
+        runtimeFeature.addOnDisposeHandler(async () => {
+            await devServerRuntime.api.devServerActions.close();
+        }, devServerEnv.env);
+
+        let serverListening = false;
+        devServerRuntime.api.serverListeningHandlerSlot.register(() => {
+            serverListening = true;
+        });
+        await waitFor(
+            () => {
+                expect(serverListening).to.eql(true);
```

Comment:
Changed to to a cleaner implementation, using the exiting featues' API, since the feature isn't an event emitter I can't once on it.

await new Promise((resolve) => {
            runtimeFeature?.api.serverListeningHandlerSlot.register(() => {
                resolve();
            });
        });

runNodeEnvironment resolves before server is listening since it resolve when the feature starts it's "run" phase, not when it finishes. It seems a bit odd to me, but it is far from the scope of this PR, and requires a lot more investigation understanding what are the implications of changing it, and whether is should even be done. @AviVahl

Reply:
Rersolved over conversation